perf/ring_test.go

```diff
@@ -62,3 +64,44 @@ func makeRing(size, offset int) *ringReader {
 
 	return newRingReader(&meta, ring)
 }
+
+func TestPerfEventRing(t *testing.T) {
```

Comment:
I kind of expected you to test mmapBufferSize here, because you made it a separate function. That might save you some boilerplate?

Reply:
It does, but testing `mmapBufferSize` is hard because it depends on `os.Getpagesize()`, which varies. It also doesn't check that the sizes are actually accepted by the kernel, I figured testing the kernel is happy was more important. But I'm happy to change it if you like :)

Reply:
Good point. Maybe add `pageSize int` as an argument? Always testing the whole hog seems nice on the surface, but it means that making changes to newPerfEvent ripple out to the tests a lot.

Reply:
I feel we should actually test the kernel accepts these combinations, and this seems like the best place to do it.
But if you'd rather just have a test for `perfBufferSize()`, I'll change it.